File: solutions/typescript/proverb/1/proverb.ts
export function proverb(...nouns: string[]): string {
  let result = "";
  for (let i: number = 0; i < nouns.length - 1; i++){
    result += `For want of a ${nouns[i]} the ${nouns[i + 1]} was lost.\n`;
  }
  result += `And all for the want of a ${nouns[0]}.`;

  return result;
}

File: solutions/python/proverb/1/proverb.py
def proverb(*input_data, qualifier):
    result = []

    if len(input_data) > 1:
        for (first, second) in list(zip(input_data, input_data[1:])):
            result.append(f"For want of a {first} the {second} was lost.")

    if len(input_data) > 0:
        qualified = f"{qualifier} {input_data[0]}" if qualifier else input_data[0]
        result.append(f"And all for the want of a {qualified}.")

    return result
